### ISSUE TITLE
Setup CI build to test CMake+Install on Linux.

### ISSUE THREAD

**Issue by coryan** [googleapis/google-cloud-cpp-spanner#23]
We need a build for macOS to verify the code compiles, can be installed, and once installed it is usable. The build should:

- [ ] Compile + install the client library.
- [ ] Run the unit tests for the client library.
- [ ] Compile a small program that depends on the client library.
- [ ] The program should use all the "clients" (e.g. if we have separate headers for "transaction client" vs. "regular client" vs. "admin client".
- [ ] Run the small program against production.

**Comment by remyabel** [comment]
>  Run the small program against production.

Can you clarify what's wanted here? I'm guessing something like:

```shell
if [[ ${RUN_INTEGRATION_TESTS} == "yes" ]]; then
# ...
./ci/test-install/spanner_install_test ${GOOGLE_CLOUD_PROJECT} \
    ${GOOGLE_CLOUD_CPP_SPANNER_INSTANCE} foo-bar
# ...
```

**Comment by coryan** [comment]
Yes, something like that. It could perform a read-only operation, such as `GetDatabase()` or `ListDatabases()`, when we have implemented them, of course.

Note that for this to work the Docker image running this must get the GCP credentials, which require some tweaking of the build configuration.
